sql/analyzer/resolve_tables.go

```diff
@@ -321,16 +306,28 @@ func validateDropTables(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope,
 
 	// validates that each table in DropTable is ResolvedTable and each database of
 	// each table is TableDropper (each table can be of different database later on)
+	var resolvedTables []sql.Node
 	for _, table := range dt.Tables {
-		rt, ok := table.(*plan.ResolvedTable)
-		if !ok {
-			return nil, transform.SameTree, plan.ErrUnresolvedTable.New(rt.String())
-		}
-		_, ok = rt.Database.(sql.TableDropper)
-		if !ok {
-			return nil, transform.SameTree, sql.ErrDropTableNotSupported.New(rt.Database.Name())
+		switch t := table.(type) {
+		case *plan.ResolvedTable:
+			_, ok = t.Database.(sql.TableDropper)
+			if !ok {
+				return nil, transform.SameTree, sql.ErrDropTableNotSupported.New(t.Database.Name())
+			}
+			resolvedTables = append(resolvedTables, table)
+		case *plan.UnresolvedTable:
+			if !dt.IfExists() {
+				return nil, transform.SameTree, sql.ErrUnknownTable.New(t.String())
+			}
+		case *plan.SubqueryAlias:
+			return nil, transform.SameTree, sql.ErrUnknownTable.New(t.Name())
+		default:
+			// TODO: try to get the name used for the error rather than the node plan string
+			return nil, transform.SameTree, sql.ErrUnknownTable.New(table.String())
```

Comment:
Would `analyzer.getTableName()` do the trick here?

Reply:
perfect!!